### ISSUE TITLE
parse_sleeptimes - roundvalue must be a whole number

### ISSUE THREAD

**Issue by lukestrickland** [humanfactors/FIPS#9]
I wanted to use a small roundvalue to check an integral against a numerical approximation. I tried to set roundvalue to a decimal, e.g.,0.5 to get predictions every 30 seconds, but I encountered an error. I also get an error for decimal roundvalues bigger than 1 (e.g., roundvalue=2.5)

I think at least we should have an error message explaining that roundvalue must be a whole number.

```r
library(FIPS)
library(tidyverse)
# Simulation start date time (i.e., when you want first predictions to begin)
simulation.start = lubridate::ymd_hms('2018-05-01 00:00:00', tz = "Australia/Perth")
# Simulation end date time (i.e., when you want predictions to end)
simulation.end = lubridate::ymd_hms('2018-05-09 21:00:00', tz = "Australia/Perth")
# Sleep times, 5 hours per night, starting 23:00
example.sleeptimes <- tibble::tibble(
  sleep.start = seq(
    from = lubridate::ymd_hms('2018-05-01 02:00:00', tz = "Australia/Perth"),
    to = lubridate::ymd_hms('2018-05-09 16:00:00', tz = "Australia/Perth"),
    by = '24 hours'),
  sleep.end = sleep.start + lubridate::dhours(3),
  sleep.id = rank(sleep.start))

simulated.dataframe = parse_sleeptimes(
  sleeptimes = example.sleeptimes,
  series.start = simulation.start,
  series.end = simulation.end,
  sleep.start.col = "sleep.start",
  sleep.end.col = "sleep.end",
  sleep.id.col = "sleep.id",
  roundvalue = 0.5)

```

---

**System Information**

```
sessionInfo()
R version 3.6.2 (2019-12-12)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18363)

Matrix products: default

locale:
[1] LC_COLLATE=English_Australia.1252  LC_CTYPE=English_Australia.1252   
[3] LC_MONETARY=English_Australia.1252 LC_NUMERIC=C                      
[5] LC_TIME=English_Australia.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] forcats_0.5.0   stringr_1.4.0   dplyr_1.0.2     purrr_0.3.4     readr_1.3.1    
 [6] tidyr_1.1.2     tibble_3.0.3    ggplot2_3.3.2   tidyverse_1.3.0 FIPS_0.1.1     

loaded via a namespace (and not attached):
 [1] tidyselect_1.1.0 haven_2.2.0      lattice_0.20-38  colorspace_1.4-1
 [5] vctrs_0.3.4      generics_0.0.2   utf8_1.1.4       rlang_0.4.7     
 [9] pillar_1.4.6     glue_1.4.2       withr_2.3.0      DBI_1.1.0       
[13] dbplyr_1.4.3     modelr_0.1.7     readxl_1.3.1     lifecycle_0.2.0 
[17] munsell_0.5.0    gtable_0.3.0     cellranger_1.1.0 rvest_0.3.5     
[21] labeling_0.3     fansi_0.4.1      broom_0.5.6      Rcpp_1.0.5      
[25] backports_1.1.10 scales_1.1.1     checkmate_2.0.0  jsonlite_1.6.1  
[29] farver_2.0.3     fs_1.4.1         hms_0.5.3        digest_0.6.25   
[33] stringi_1.4.6    grid_3.6.2       cli_2.0.2        tools_3.6.2     
[37] magrittr_1.5     crayon_1.3.4     pkgconfig_2.0.3  ellipsis_0.3.1  
[41] xml2_1.2.2       reprex_0.3.0     lubridate_1.7.9  assertthat_0.2.1
[45] httr_1.4.1       rstudioapi_0.11  R6_2.4.1         nlme_3.1-142    
[49] compiler_3.6.2  
```

**Comment by humanfactors** [comment]
Thanks for spotting this. 

I believe I only implemented FIPS to have a minimum resolution of 1 minute (as you have detected), as seconds would require an additional layer of complexity for rounding values (i.e., we would need to be parsing seconds). Currently, both `round_times` and `expand_sleep_series` would require some hefty changes to accommodate seconds. So I'm not sure it's worth at this stage implementing.

The `sleepwakevector` format should implicitly support sub one minute epochs, but I would need to test to confirm this. 

For now I'll add a checker and handle this case, and direct the user to the other format.

**Comment by lukestrickland** [comment]
Thanks that seems totally sensible. I think we shouldn't make hefty changes to get resolution below minutes... just simple check as you've suggested

**Comment by humanfactors** [comment]
Addressed in #10 
Closed.
